src/DynamoCoreWpf/Views/Search/LibraryView.xaml.cs

```diff
@@ -180,19 +184,48 @@ private void ExpandCategory(TreeViewItem sender, NodeCategoryViewModel selectedC
             foreach (var categoryToBeCollapsed in allExpandedCategories)
             {
                 // If class button was clicked.
-                if (categoryToBeExpanded == null)
+                if (selectedClass.SubCategories.Count == 0)
                 {
                     var categoryClasses = categoryToBeCollapsed.Items[0] as ClassesNodeCategoryViewModel;
                     // Ensure, that this class is not part of current category.
                     if (categoryClasses != null)
-                        categoryToBeCollapsed.IsExpanded = categoryClasses.Items.Contains(selectedClass);
+                    {
+                        if (categoryClasses.Items.Contains(selectedClass))
+                        {
+                            categoryToBeCollapsed.IsExpanded = true;
+                            // Selected class has been found! Collapse all other sub categories.
+                            foreach (var ele in categoryToBeCollapsed.SubCategories.Where(cat => cat.IsExpanded == true))
+                                ele.IsExpanded = false;
+                        }
+                        else
```

Comment:
Please don't omit the curly brackets for the above `else` statement since it has a multi-line body.